src/main/java/com/coveo/pushapiclient/DocumentBuilder.java

```diff
@@ -372,10 +375,18 @@ public String marshal() {
   public JsonObject marshalJsonObject() {
     this.generatePermanentId();
 
-    JsonObject jsonDocument = new Gson().toJsonTree(this.document).getAsJsonObject();
+    Gson gson = new Gson();
+    JsonObject jsonDocument = gson.toJsonTree(this.document).getAsJsonObject();
     this.document.metadata.forEach(
         (key, value) -> {
-          jsonDocument.add(key, new Gson().toJsonTree(value));
+          if (value instanceof Collection) {
+            jsonDocument.add(
+                key, new JsonPrimitive(StringUtils.join(((Collection<?>) value).toArray(), ";")));
```

Comment:
my biggest worry here is that this could be a breaking change for clients using the current version. If they require the , as a seperator rather than the ; then this could cause them a problem if they upgrade.

Reply:
As we discussed previously any change made in sdk which will change any part of the result can be breaking for the clients. Other way is to hard code it for each specific cases in specific value providers.

Reply:
Which field would this change impact? Some ideas to mitigate the risk could be:

1. Look at the tailgate database to see what customers are sending today?
2. Is it possible to make the feature work with a `,`? If not, then maybe this has always been a bug.
3. Release the sdk under a new major version to communicate that it contains a breaking change.